core/internal/wbvalue/image.go

```diff
@@ -23,6 +31,54 @@ type Image struct {
 	Height int
 }
 
+// ImageFromData returns an Image from encoded data.
+func ImageFromData(
+	width int,
+	height int,
+	encodedData []byte,
+) (Image, error) {
+	var pngData []byte
+
+	// Special case for PNG data which we can send directly.
+	if len(encodedData) >= 8 && slices.Equal(
+		encodedData[:8],
+		[]byte{0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A},
+	) {
+		pngData = encodedData
+	} else {
```

Comment:
wait we always need to convert it into png?

Reply:
hmm... good point, it might be possible to send the data directly for other formats. I believe our legacy TB integration always encodes to PNG or GIF.

Reply:
Interesting, for some reason I remember that our ui supports more encoding. You can check in the image media file

Reply:
Our UI does support other encodings; our TB integration only ever _outputs_ PNG or GIF.

Reply:
Yeah but we are going to render it in the workspace tab so what matters is what our UI supports.

For the tb tab, it just reads the tb events directly

Reply:
I'm saying the legacy Python code only ever encoded images to PNG or GIF, regardless of the format used in the tfevent file.

Anyway, I updated this to pass-through the format and encoded data instead of re-encoding to PNG.

Reply:
Sounds good